assets/src/design-system/components/toggle/index.js

```diff
@@ -0,0 +1,190 @@
+    position: absolute;
+    top: -${BORDER_WIDTH}px;
+    left: -${BORDER_WIDTH}px;
+    height: ${TOGGLE_HEIGHT}px;
+    width: ${TOGGLE_WIDTH}px;
+    background-color: transparent;
+    border-radius: ${theme.borders.radius.x_large};
+    border: ${BORDER_WIDTH}px solid ${theme.colors.border.defaultNormal};
+    pointer-events: none;
+    transition: all 0.3s;
+  `
+);
+
+const Circle = styled.span(
+  ({ theme }) => css`
+    pointer-events: none;
+
+    :after {
+      content: '';
+      position: absolute;
+      top: ${CIRCLE_INITIAL_POSITION}px;
+      left: ${CIRCLE_INITIAL_POSITION}px;
+      height: ${CIRCLE_DIAMETER}px;
+      width: ${CIRCLE_DIAMETER}px;
+      background-color: ${theme.colors.fg.secondary};
+      border-radius: ${theme.borders.radius.round};
+      cursor: pointer;
+      transition: background-color 0.3s, border-color 0.3s, left 0.15s;
+    }
+  `
+);
+
+const StyledCheckmark = styled(Checkmark)(
+  ({ theme }) => css`
+    position: absolute;
+    width: ${ICON_WIDTH}px;
+    top: ${ICON_TOP_POSITION}px;
+    left: ${ICON_LEFT_POSITION}px;
+    z-index: 1;
+    opacity: 0;
+    transition: opacity 0.15s;
+    pointer-events: none;
+
+    path {
+      fill: ${theme.colors.standard.white};
+    }
+  `
+);
+
+const ToggleContainer = styled.div(
+  ({ theme }) => css`
+    position: relative;
+    height: ${TOGGLE_HEIGHT}px;
+    width: ${TOGGLE_WIDTH}px;
+    background-color: ${theme.colors.bg.primary};
+    border-radius: ${theme.borders.radius.x_large};
+
+    input[type='checkbox'] {
+      position: absolute;
+      top: -${BORDER_WIDTH / 2}px;
+      left: -${BORDER_WIDTH / 2}px;
+      height: ${TOGGLE_HEIGHT + BORDER_WIDTH}px;
+      width: ${TOGGLE_WIDTH + BORDER_WIDTH}px;
+      margin: 0;
+      opacity: 0;
+      cursor: pointer;
+
+      :disabled {
+        ~ ${Background} {
+          border-color: ${theme.colors.fg.disable};
+        }
+
+        :checked ~ ${Background} {
+          background-color: ${theme.colors.fg.disable};
+          border-color: ${theme.colors.fg.disable};
+        }
+
+        ~ ${Circle}:after {
+          background-color: ${theme.colors.fg.disable};
+        }
+
+        :checked ~ ${Circle}:after {
+          background-color: ${theme.colors.bg.secondary};
+        }
+
+        ~ ${StyledCheckmark} path {
+          fill: ${theme.colors.interactiveFg.disable};
+        }
+      }
+
+      &.${FOCUS_VISIBLE_SELECTOR} {
+        ~ ${Background} {
+          outline: none;
+          box-shadow: 0 0 0 5px ${theme.colors.bg.primary},
+            0 0 0 7px ${theme.colors.border.focus};
+        }
+      }
+
+      :checked {
+        ~ ${Background} {
+          background-color: ${theme.colors.interactiveBg.positivePress};
+          border-color: ${theme.colors.interactiveBg.positivePress};
+        }
+
+        ~ ${Circle}:after {
+          left: ${CIRCLE_FINAL_POSITION}px;
+          background-color: ${theme.colors.interactiveBg.positiveNormal};
+        }
```

Comment:
This has no perf issues, so it's certainly not needed, but we could animate a `transform` instead of `left` here for increased perf. No need to change this if you're about to merge in cus it is working fine rn, but something to keep in mind as you animate position in the future
```
transform: translateX(${CIRCLE_FINAL_POSITION}px);
```
It allows the browser to skip some extra steps in the rendering process while achieving the same net result:
https://www.html5rocks.com/en/tutorials/speed/high-performance-animations/

Reply:
you're a rockstar